src/main/java/me/tomsdevsn/hetznercloud/HetznerCloudAPI.java

```diff
@@ -1,56 +1,47 @@
 package me.tomsdevsn.hetznercloud;
 
+import com.fasterxml.jackson.databind.DeserializationFeature;
+import com.fasterxml.jackson.databind.ObjectMapper;
 import me.tomsdevsn.hetznercloud.exception.InvalidParametersException;
 import me.tomsdevsn.hetznercloud.objects.general.PlacementGroupType;
 import me.tomsdevsn.hetznercloud.objects.pagination.PaginationParameters;
 import me.tomsdevsn.hetznercloud.objects.request.*;
 import me.tomsdevsn.hetznercloud.objects.response.*;
-import org.springframework.http.HttpEntity;
-import org.springframework.http.HttpHeaders;
-import org.springframework.http.HttpMethod;
-import org.springframework.http.MediaType;
-import org.springframework.http.converter.HttpMessageConverter;
-import org.springframework.http.converter.json.MappingJackson2HttpMessageConverter;
-import org.springframework.web.client.RestTemplate;
-import org.springframework.web.util.UriComponentsBuilder;
-
-import java.text.DateFormat;
-import java.text.SimpleDateFormat;
-import java.util.*;
+import okhttp3.MediaType;
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+import okhttp3.RequestBody;
+
+import java.io.IOException;
+import java.util.Optional;
 import java.util.concurrent.TimeUnit;
 
 public class HetznerCloudAPI {
 
     private static final String API_URL = "https://api.hetzner.cloud/v1";
 
-    private final String token;
+    private final OkHttpClient client;
 
-    private final HttpEntity<String> httpEntity;
-    private final HttpHeaders httpHeaders;
-    private final RestTemplate restTemplate;
+    private final String hcloudToken;
+    private final ObjectMapper objectMapper;
 
     /**
      * Initial method to use the API
      *
      * @param token API-Token for Hetzner Cloud API
      *              The API token can be created within the Hetzner Cloud Console
      */
-    public HetznerCloudAPI(String token) {
-        this.token = token;
+    public HetznerCloudAPI(String hcloudToken) {
 
-        List<HttpMessageConverter<?>> messageConverters = new ArrayList<>();
-        MappingJackson2HttpMessageConverter converter = new MappingJackson2HttpMessageConverter();
-        converter.setSupportedMediaTypes(Collections.singletonList(MediaType.APPLICATION_JSON));
-        messageConverters.add(converter);
+        if (hcloudToken == null || hcloudToken.isBlank()) {
```

Comment:
Unnecessary brackets.

Reply:
You mean the curly braces? Yes they are optional, but I would argue from my personal experience that the amount of bugs caused by missing braces I have seen in codebases over the years (most notably example would perhaps to a certain degree be https://www.imperialviolet.org/2014/02/22/applebug.html) by a huge margin outweighs any advantage gained by omitting those braces.